smack-experimental/src/main/java/org/jivesoftware/smackx/httpauthorizationrequest/HTTPAuthorizationRequestManager.java

```diff
@@ -0,0 +1,217 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.httpauthorizationrequest;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.WeakHashMap;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jivesoftware.smack.Manager;
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.filter.AndFilter;
+import org.jivesoftware.smack.filter.FromTypeFilter;
+import org.jivesoftware.smack.filter.MessageTypeFilter;
+import org.jivesoftware.smack.filter.StanzaExtensionFilter;
+import org.jivesoftware.smack.filter.StanzaFilter;
+import org.jivesoftware.smack.iqrequest.AbstractIqRequestHandler;
+import org.jivesoftware.smack.iqrequest.IQRequestHandler;
+import org.jivesoftware.smack.packet.IQ;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.packet.Message.Body;
+import org.jivesoftware.smack.packet.MessageBuilder;
+import org.jivesoftware.smack.packet.StanzaBuilder;
+import org.jivesoftware.smack.packet.StanzaError;
+import org.jivesoftware.smackx.httpauthorizationrequest.element.ConfirmExtension;
+import org.jivesoftware.smackx.httpauthorizationrequest.packet.ConfirmIQ;
+
+import org.jxmpp.jid.DomainBareJid;
+import org.jxmpp.jid.Jid;
+
+/**
+ * The HTTP Request manager for HTTP Request. The manager handles both the HTTP Request via:
+ * IQ : when the request is via EntityFullJid
+ * Message: When request is via EntityBareJid
+ *
+ * XEP-0070: Verifying HTTP Requests via XMPP (1.0.1 (2016-12-09))
+ *
+ * @see HTTPAuthorizationRequestListener on callback
+ */
+public final class HTTPAuthorizationRequestManager extends Manager {
+    private static final StanzaFilter MESSAGE_FILTER = new AndFilter(
+            MessageTypeFilter.NORMAL_OR_CHAT,
+            new StanzaExtensionFilter(ConfirmExtension.ELEMENT, ConfirmExtension.NAMESPACE)
+    );
+
+    private static final StanzaFilter INCOMING_MESSAGE_FILTER = new AndFilter(
+            MESSAGE_FILTER,
+            FromTypeFilter.DOMAIN_BARE_JID
+    );
+
+    private static final Logger LOGGER = Logger.getLogger(HTTPAuthorizationRequestManager.class.getName());
+
+    private static final Map<XMPPConnection, HTTPAuthorizationRequestManager> INSTANCES = new WeakHashMap<>();
+
+    private final Set<HTTPAuthorizationRequestListener> incomingListeners = new CopyOnWriteArraySet<>();
+
+    private ConfirmExtension mConfirmExtension;
```

Comment:
The current design of the Manager is far from ideal, it is racy, allows at most one outstanding verification, and you never know what you accept or reject.

If I understand the protocol correctly, then the manager could simply provide a method
```
void acceptId(URI url, String method, String id)
```
while keeping a Set of all outstanding IDs that it will accept. That way, the listeners simply look if the incoming ID is in the set, and, if so, remove the ID from the set and accept, otherwise reject.

Reply:
Add the following to support multiple instances of auth requests:
a. private static final Map<String, Object> mAuthRequests = new HashMap<>()
b. public void acceptId(String id)
c. public void rejectId(String id)